packages/engine-server/src/markdown/remark/noteRefsV2.ts

```diff
@@ -260,14 +262,28 @@ function convertNoteRef(opts: ConvertNoteRefOpts): {
       data: "",
     };
   }
-  let { prettyRefs, wikiLinkOpts } = compilerOpts;
+  const { wikiLinkOpts } = compilerOpts;
+  let { prettyRefs } = compilerOpts;
   if (refLvl >= MAX_REF_LVL) {
     return {
       error: new DendronError({ message: "too many nested note refs" }),
       data,
     };
   }
 
+  // The note that contains this reference might override the pretty refs option for references inside it.
+  const containingNote = NoteUtils.getNoteByFnameFromEngine({
+    fname,
+    vault,
+    engine,
+  });
+  if (
+    containingNote?.custom.usePrettyRefs !== undefined &&
```

Comment:
a few points:
- we now have a lot of reserved frontmatter keys, we want to scope them to some top level modules
- we already have a option for pretty ref in new config, lets use that `enablePrettyRef`
- with config migration, we are moving all config keys into modules, so lets also do that

for now, since we haven't migrated the `site` namespace to `publish`, i would recommend using the `global` namepsace. so the end result would look like this:

```yml
config:
  global:
    enablePrettyRefs
```

you can see how we currently do it for childLinks here -> [getEnableChildLinks](https://github.com/dendronhq/dendron/blob/93f256103fc8d99ca2777b38fe5c26d074eb7de6/packages/common-all/src/utils.ts#L666)